### PR TITLE
Add PortAssigner

### PR DESCRIPTION
**Pull request by chrisrohr**
Closes #3

### CODE REVIEW

File: src/main/java/org/kiwiproject/dropwizard/util/startup/PortAssigner.java [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+        return https;
+    }
+
+    private void assignNonSecureDynamicPorts() {
+        LOG.debug("Insecure (http): modify Dropwizard HTTP app/admin connectors using dynamic ports");
+
+        var usedPorts = new HashSet<Integer>();
+
+        int connectorPort = findFreePort(usedPorts);
```
Comment (sleberknight) [inline src/main/java/org/kiwiproject/dropwizard/util/startup/PortAssigner.java:125]: Suggest renaming this to `applicationPort`.

File: src/main/java/org/kiwiproject/dropwizard/util/startup/PortAssigner.java [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+import java.util.List;
+import java.util.Optional;
+import java.util.OptionalInt;
+import java.util.Set;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.function.IntSupplier;
+import java.util.stream.IntStream;
+
+/**
+ * Finds open ports and sets up the application and admin connectors with those ports. There are a couple of opinionated decisions here:
```
Comment (sleberknight) [inline src/main/java/org/kiwiproject/dropwizard/util/startup/PortAssigner.java:32]: Do we imply another "opinionated decision" by assuming there is only _one_ application and _one_ admin port? Or is that not true? What happens if there is a secure _and_ non-secure port for both application and admin?

**Review by sleberknight — CHANGES_REQUESTED** [review]

File: src/main/java/org/kiwiproject/dropwizard/util/startup/PortAssigner.java (hunk 2 above)
Comment (chrisrohr) [reply src/main/java/org/kiwiproject/dropwizard/util/startup/PortAssigner.java:32]: Yes, that is an opinionated decision that we only have one of each.  This does not let more than one or both secure and non-secure.

**Comment by chrisrohr** [comment]
Let me add one more test to get this to 100%

File: src/main/java/org/kiwiproject/dropwizard/util/startup/PortAssigner.java (hunk 2 above)
Comment (chrisrohr) [reply src/main/java/org/kiwiproject/dropwizard/util/startup/PortAssigner.java:32]: I added another line in the javadoc to indicate this decision

**Comment by chrisrohr** [comment]
> Let me add one more test to get this to 100%

This is done now

**Review by sleberknight — APPROVED** [review]

**Review by sleberknight — APPROVED** [review]

**Pull request merged**